srml/rolling-window/src/mock.rs

```diff
@@ -0,0 +1,313 @@
+		self.minimum_validator_count = count;
+		self
+	}
+	pub fn fair(mut self, is_fair: bool) -> Self {
+		self.fair = is_fair;
+		self
+	}
+	pub fn num_validators(mut self, num_validators: u32) -> Self {
+		self.num_validators = Some(num_validators);
+		self
+	}
+	pub fn set_associated_consts(&self) {
+		EXISTENTIAL_DEPOSIT.with(|v| *v.borrow_mut() = self.existential_deposit);
+	}
+	pub fn build(self) -> runtime_io::TestExternalities<Blake2Hasher> {
+		self.set_associated_consts();
+		let (mut t, mut c) = system::GenesisConfig::default().build_storage::<Test>().unwrap();
+		let balance_factor = if self.existential_deposit > 0 {
+			256
+		} else {
+			1
+		};
+
+		let num_validators = self.num_validators.unwrap_or(self.validator_count);
+		let validators = (0..num_validators)
+			.map(|x| ((x + 1) * 10) as u64)
+			.collect::<Vec<_>>();
+
+		let _ = balances::GenesisConfig::<Test>{
+			balances: vec![
+					(1, 10 * balance_factor),
+					(2, 20 * balance_factor),
+					(3, 300 * balance_factor),
+					(4, 400 * balance_factor),
+					(10, balance_factor),
+					(11, balance_factor * 1000),
+					(20, balance_factor),
+					(21, balance_factor * 2000),
+					(30, balance_factor),
+					(31, balance_factor * 2000),
+					(40, balance_factor),
+					(41, balance_factor * 2000),
+					(100, 2000 * balance_factor),
+					(101, 2000 * balance_factor),
+			],
+			vesting: vec![],
+		}.assimilate_storage(&mut t, &mut c);
+
+		let stake_21 = if self.fair { 1000 } else { 2000 };
+		let stake_31 = if self.validator_pool { balance_factor * 1000 } else { 1 };
+		let status_41 = if self.validator_pool {
+			StakerStatus::<AccountId>::Validator
+		} else {
+			StakerStatus::<AccountId>::Idle
+		};
+		let nominated = if self.nominate { vec![11, 21] } else { vec![] };
+		let _ = GenesisConfig::<Test>{
+			current_era: 0,
+			stakers: vec![
+				(11, 10, balance_factor * 1000, StakerStatus::<AccountId>::Validator),
+				(21, 20, stake_21, StakerStatus::<AccountId>::Validator),
+				(31, 30, stake_31, StakerStatus::<AccountId>::Validator),
+				(41, 40, balance_factor * 1000, status_41),
+				// nominator
+				(101, 100, balance_factor * 500, StakerStatus::<AccountId>::Nominator(nominated))
+			],
+			validator_count: self.validator_count,
+			minimum_validator_count: self.minimum_validator_count,
+			session_reward: Perbill::from_millionths((1000000 * self.reward / balance_factor) as u32),
+			offline_slash: Perbill::from_percent(5),
+			current_session_reward: self.reward,
+			offline_slash_grace: 0,
+			invulnerables: vec![],
+		}.assimilate_storage(&mut t, &mut c);
+
+		let _ = srml_session::GenesisConfig::<Test> {
+			keys: validators.iter().map(|x| (*x, UintAuthorityId(*x))).collect(),
+		}.assimilate_storage(&mut t, &mut c);
+
+		let mut ext = t.into();
+		runtime_io::with_externalities(&mut ext, || {
+			let validators = Session::validators();
+			SESSION.with(|x|
+				*x.borrow_mut() = (validators.clone(), HashSet::new())
+			);
+		});
+		ext
+	}
+}
```

Comment:
I wonder if we really need to write all of this, looks like too much work.

Reply:
it's definitely longer than other SRML mocks, not sure why that is

Reply:
copy-pasted from `srml-staking/src/mock`

Reply:
I don't think subtelties of config of staking should copy paste here, make the config of staking mock an ok default for your tests

also I'm not even sure you need staking for your test do you ?

Reply:
yepp, it's probably not needed